### ISSUE TITLE
MPNN what each special arg carry 

### ISSUE THREAD

**Issue by BAlmeidaS** [pyg-team/pytorch_geometric#9476]
### 📚 Describe the documentation issue

There is an inspector instance inside MPNN that [removes any argument with the same name](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/message_passing.py#L146-L151) as the ones in [special_args](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/message_passing.py#L101:L104).

**Comment by rusty1s** [comment]
I agree that we need better documentation around these special arguments that are available as function parameters in `message` and `aggregate`. Happy to answer any questions in the meantime.

**Comment by rajveer43** [comment]
I would like to contribute to this.
